### PR TITLE
Trash

### PR DESCRIPTION
**Pull request by harshalini**
Added Trash functionalities in notes app
Now a user can 
1) Move added notes to trash
2) Restore note from trash
3) Move archived note to trash
4) Delete a note from trash
after signing up or logging in the app

Preview -
https://luxury-pithivier-a56288.netlify.app/

### CODE REVIEW

File: src/pages/trash.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { Sidebar } from "../components/allComp";
+import { useTrash } from "../context/trashContext";
+export const TrashPage = () => {
+    const { noteTrash, RestoreTrashNote, DeleteNoteFromTrash } = useTrash();
+    return (
+        <div>
+            <Sidebar />
```
Comment (Kothalkarpriya) [inline src/pages/trash.js:7]: If you want you can fix the sidebar directly into the routes path or home page

**Review by Kothalkarpriya — APPROVED** [review]
Good work! code is clean and understandable, You can merge the changes

File: src/context/trashContext.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { createContext } from "react";
+import { useState } from "react";
+import { useContext, useEffect } from "react";
```
Comment (kgajriaAgarwal) [inline src/context/trashContext.js]: suggestion: can keep this in one line!

File: src/context/trashContext.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    (async () => {
+      try {
+        const {
+          data: { trash }
+        } = await axios.get('/api/trash', {
+          headers: { authorization: authToken }
+        });
+        setNoteTrash(trash);
+      } catch (error) {
+        console.log(error);
```
Comment (kgajriaAgarwal) [inline src/context/trashContext.js:19]: remove console and implement toasts to show error and success messages

File: src/context/trashContext.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    })();
+  }, [noteTrash]);
+
+  const TrashNote = async (noteTrash, _id) => {
+    const getTrash = await AddNoteToTrash(noteTrash, _id)
+    console.log(getTrash.data.trash)
```
Comment (kgajriaAgarwal) [inline src/context/trashContext.js]: remove console.log here

File: src/context/trashContext.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  const DeleteNoteFromTrash = async(noteTrash, _id) => {
+    const response = await DeleteFromTrash(noteTrash, _id)
+    setNoteTrash(response.data.trash)
+  }
+
+  const ArchiveNoteToTrash = async(noteTrash, _id) => {
+    const response = await ArchiveToTrash(noteTrash, _id)
+    setNoteTrash(response.data.trash)
+    console.log(response.data.archives)
```
Comment (kgajriaAgarwal) [inline src/context/trashContext.js]: remove console.log

File: src/noteServices/trashService.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            note: data
+        },
+            {
+                headers: { authorization: authToken }
+            })
+        console.log(res)
+        return res;
+    }
+    catch (error) {
+        console.log("error occured", error)
```
Comment (kgajriaAgarwal) [inline src/noteServices/trashService.js:50]: use toast to show error and success messages

**Review by kgajriaAgarwal — CHANGES_REQUESTED** [review]
remove console from various places.
Implement alert/ toasts in your app to show error or success messges.
The code is clean and understandable

File: src/noteServices/trashService.js (hunk 6 above)
Comment (harshalini) [reply src/noteServices/trashService.js:50]: Yes I will do it later on

File: src/pages/trash.js (hunk 1 above)
Comment (harshalini) [reply src/pages/trash.js:7]: But the sidebar is not in the home page, thats why I kept it here

**Comment by harshalini** [comment]
> Good work! code is clean and understandable, You can merge the changes

Thank you

**Pull request merged**